### PR TITLE
Improve docs deployment

### PR DESCRIPTION
**Pull request by anbraten**
As preparation for #302 this PR introduces a new way of deploying the docs which allows us to exclude (not delete / replace) some files from the github page.

### CODE REVIEW

File: .woodpecker/docs.yml [hunk 1]
```diff
@@ -2,39 +2,40 @@ clone:
...
       - echo "$BOT_PRIVATE_KEY" > $HOME/.ssh/id_rsa
       - chmod 0600 $HOME/.ssh/id_rsa
       - git config --global user.email "woodpecker-bot@obermui.de"
       - git config --global user.name "woodpecker-bot"
-      - cd docs/
-      - yarn install --frozen-lockfile
-      - yarn deploy
+      - git clone -b master --single-branch git@github.com:woodpecker-ci/woodpecker-ci.github.io.git /repo
+      # copy all docs files and delete all old ones, but leave CNAME and index.yaml untouched
+      - rsync -r --exclude .git --exclude CNAME --exclude index.yaml --delete docs/build/ /repo
```
Comment (6543) [inline .woodpecker/docs.yml:30]:
```suggestion
      - rsync -r --exclude .git --exclude README.md --exclude index.yaml --delete docs/build/ /repo
```

Comment (6543) [reply .woodpecker/docs.yml:30]: what is CNAME for?

Comment (anbraten) [reply .woodpecker/docs.yml:30]: CNAME is the CNAME file you created for the domain for the github-page.

**Comment by 6543** [comment]
there is an issue somewhere: https://wp.laszlo.cloud/woodpecker-ci/woodpecker/423/4

-> there should be a diff but there is none

**Comment by anbraten** [comment]
> there is an issue somewhere: https://wp.laszlo.cloud/woodpecker-ci/woodpecker/423/4
> 
> -> there should be a diff but there is none
> 
> to play around just push to **test** branch

I think the test statement you changed was wrong on the test branch. To me it looks fine now.

**Review by 6543 — APPROVED** [review]

**Pull request merged**